File: XPile 3-17/Programs/test_dowhile.js

function main() {
    let count = 0;
    do {
        count = (count + 1);
    } while ((count < 5));
    console.log(count);
    let x = 10;
    do {
        x = (x + 1);
    } while ((x < 5));
    console.log(x);
    let n = 0;
    do {
        n = (n + 1);
        if ((n === 3)) {
            break;
        }
    } while ((n < 10));
    console.log(n);
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();

File: XPile 3-17/Programs/test_dowhile.py
from dataclasses import dataclass, field

def __ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class __ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)


def main():
    count = 0
    while True:
        count = (count + 1)
        if not ((count < 5)):
            break
    print(count)
    x = 10
    while True:
        x = (x + 1)
        if not ((x < 5)):
            break
    print(x)
    n = 0
    while True:
        n = (n + 1)
        if (n == 3):
            break
        if not ((n < 10)):
            break
    print(n)
    return 0


if __name__ == "__main__":
    main()
